### ISSUE TITLE
Asynchronous player kick!

### ISSUE THREAD

**Issue by cmp71** [SkriptLang/Skript#5831]
### Skript/Server Version

```markdown
[09:50:18 INFO]:  [ Skript ]  Skript's aliases can be found here:  https://github.com/SkriptLang/skript-aliases 
[09:50:18 INFO]:  [ Skript ]  Skript's documentation can be found here:  https://docs.skriptlang.org/ 
[09:50:18 INFO]:  [ Skript ]  Skript's tutorials can be found here:  https://docs.skriptlang.org/tutorials 
[09:50:18 INFO]:  [ Skript ]  Server Version:  git-Purpur-2014 (MC: 1.20.1) 
[09:50:18 INFO]:  [ Skript ]  Skript Version:  2.7.0-beta3 (skriptlang-github) 
[09:50:18 INFO]:  [ Skript ]  Installed Skript Addons:  None 
[09:50:18 INFO]:  [ Skript ]  Installed dependencies:   
[09:50:18 INFO]:  [ Skript ]   - GriefPrevention v16.18.1
```

### Bug Description

Attempting to kick a player if they infringe a chat filter.
The filters are in a list and checked via a loop following a chat event.
The kick command results in the error listed.
Different methods tried-
1. Kick command embedded directly in loop
2. Kick command outside the loop triggered by a flag variable
3. Kick command moved into a function, called directly from within the loop
4. Kick command moved into a function, called outside the loop
All methods result in the severe error Asynchronous player kick!

### Expected Behavior

Player that initiated the chat event and matched one of the defined filters gets kicked from the server.

### Steps to Reproduce

```
on load:
    delete {pottyMouth::*}
    add "(?i)\bb+[\W_]*[uv]+[\W_]*m" to {pottyMouth::*} # bum
    send "pottyMouth alert: Loaded %size of {pottyMouth::*}% regex filters" to console
    
on chat:
    set {_match} to false
    loop {pottyMouth::*}:
        if message partially matches "%loop-value%":
            cancel event
            log "%time%: %player% said %message%" to "pottymouth.log"
            send "pottyMouth alert: %player% said %message%" to console
            set {_match} to true
            exit loop
    if {_match} is true:
        punishPlayer(player)

function punishPlayer(player2Punish: player):
    kick {_player2Punish} because "pottyMouth"
```

1. Create a skript using the above code and load
2. Connect a player to the server and type **bum** in the chat

### Errors or Screenshots

```
[09:42:44] [Server thread/INFO]: [Skript] Reloading the config, aliases and all scripts...
[09:42:46] [Server thread/INFO]: Loaded 150508 aliases in 2525ms
[09:42:46] [Server thread/INFO]: pottyMouth alert: Loaded 59 regex filters
[09:42:46] [Server thread/INFO]: [Skript] Successfully reloaded the config, aliases and all scripts. (2549ms)
[09:43:01] [Async Chat Thread - #21/INFO]: pottyMouth alert: OtisCat said cunt
[09:43:01] [Async Chat Thread - #21/ERROR]: Thread Async Chat Thread - #21 failed main thread check: player kick
java.lang.Throwable: null
	at org.spigotmc.AsyncCatcher.catchOp(AsyncCatcher.java:15) ~[purpur-1.20.1.jar:git-Purpur-2014]
	at org.bukkit.craftbukkit.v1_20_R1.entity.CraftPlayer.kickPlayer(CraftPlayer.java:588) ~[purpur-1.20.1.jar:git-Purpur-2014]
	at ch.njol.skript.effects.EffKick.execute(EffKick.java:81) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.lang.Effect.run(Effect.java:50) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:62) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:90) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.lang.Trigger.execute(Trigger.java:52) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.lang.function.ScriptFunction.execute(ScriptFunction.java:86) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.lang.function.Function.execute(Function.java:129) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.lang.function.FunctionReference.execute(FunctionReference.java:301) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.lang.function.EffFunctionCall.execute(EffFunctionCall.java:52) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.lang.Effect.run(Effect.java:50) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:62) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:90) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.lang.Trigger.execute(Trigger.java:52) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.SkriptEventHandler.check(SkriptEventHandler.java:146) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.SkriptEventHandler.access$000(SkriptEventHandler.java:50) ~[Skript-2.7.0-beta3.jar:?]
	at ch.njol.skript.SkriptEventHandler$PriorityListener.lambda$new$0(SkriptEventHandler.java:63) ~[Skript-2.7.0-beta3.jar:?]
	at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77) ~[purpur-api-1.20.1-R0.1-SNAPSHOT.jar:git-Purpur-2014]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[purpur-api-1.20.1-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54) ~[purpur-1.20.1.jar:git-Purpur-2014]
	at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126) ~[purpur-1.20.1.jar:git-Purpur-2014]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:617) ~[purpur-api-1.20.1-R0.1-SNAPSHOT.jar:?]
	at io.papermc.paper.adventure.ChatProcessor.post(ChatProcessor.java:414) ~[purpur-1.20.1.jar:git-Purpur-2014]
	at io.papermc.paper.adventure.ChatProcessor.process(ChatProcessor.java:100) ~[purpur-1.20.1.jar:git-Purpur-2014]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.chat(ServerGamePacketListenerImpl.java:2498) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.broadcastChatMessage(ServerGamePacketListenerImpl.java:2654) ~[?:?]
	at net.minecraft.server.network.ServerGamePacketListenerImpl.lambda$handleChat$18(ServerGamePacketListenerImpl.java:2326) ~[?:?]
	at java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718) ~[?:?]
	at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! 
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! [Skript] Severe Error:
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! 
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! Something went horribly wrong with Skript.
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! This issue is NOT your fault! You probably can't fix it yourself, either.
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! You should report it at https://github.com/SkriptLang/Skript/issues. Please copy paste this report there (or use paste service).
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! This ensures that your issue is noticed and will be fixed as soon as possible.
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! 
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! Stack trace:
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! java.lang.IllegalStateException: Asynchronous player kick!
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at org.spigotmc.AsyncCatcher.catchOp(AsyncCatcher.java:16)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at org.bukkit.craftbukkit.v1_20_R1.entity.CraftPlayer.kickPlayer(CraftPlayer.java:588)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.effects.EffKick.execute(EffKick.java:81)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.lang.Effect.run(Effect.java:50)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:62)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:90)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.lang.Trigger.execute(Trigger.java:52)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.lang.function.ScriptFunction.execute(ScriptFunction.java:86)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.lang.function.Function.execute(Function.java:129)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.lang.function.FunctionReference.execute(FunctionReference.java:301)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.lang.function.EffFunctionCall.execute(EffFunctionCall.java:52)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.lang.Effect.run(Effect.java:50)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:62)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.lang.TriggerItem.walk(TriggerItem.java:90)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.lang.Trigger.execute(Trigger.java:52)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.SkriptEventHandler.check(SkriptEventHandler.java:146)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.SkriptEventHandler.access$000(SkriptEventHandler.java:50)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at Skript-2.7.0-beta3.jar//ch.njol.skript.SkriptEventHandler$PriorityListener.lambda$new$0(SkriptEventHandler.java:63)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at co.aikar.timings.TimedEventExecutor.execute(TimedEventExecutor.java:77)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at io.papermc.paper.plugin.manager.PaperEventManager.callEvent(PaperEventManager.java:54)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at io.papermc.paper.plugin.manager.PaperPluginManagerImpl.callEvent(PaperPluginManagerImpl.java:126)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:617)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at io.papermc.paper.adventure.ChatProcessor.post(ChatProcessor.java:414)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at io.papermc.paper.adventure.ChatProcessor.process(ChatProcessor.java:100)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at net.minecraft.server.network.PlayerConnection.chat(PlayerConnection.java:2498)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at net.minecraft.server.network.PlayerConnection.b(PlayerConnection.java:2654)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at net.minecraft.server.network.PlayerConnection.lambda$handleChat$18(PlayerConnection.java:2326)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     at java.base/java.lang.Thread.run(Thread.java:833)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! 
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! Version Information:
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!   Skript: 2.7.0-beta3 (latest)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     Flavor: skriptlang-github
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!     Date: 03:18:03.366646700
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!   Bukkit: 1.20.1-R0.1-SNAPSHOT
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!   Minecraft: 1.20.1
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!   Java: 17.0.7 (OpenJDK 64-Bit Server VM 17.0.7+7-Debian-1deb11u1)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#!   OS: Linux amd64 5.10.0-23-amd64
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! 
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! Server platform: Paper
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! 
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! Current node: null
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! Current item: kick {_player2Punish} (as org.bukkit.entity.Player) on account of "pottyMouth"
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! Current trigger: function punishPlayer (simple event) (pottyMouth.sk, line 76)
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! 
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! Thread: Async Chat Thread - #21
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! 
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! Language: english
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! Link parse mode: DISABLED
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! 
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! End of Error.
[09:43:01] [Async Chat Thread - #21/ERROR]: #!#! 
[09:50:18] [Server thread/INFO]: [Skript] Skript's aliases can be found here: https://github.com/SkriptLang/skript-aliases
[09:50:18] [Server thread/INFO]: [Skript] Skript's documentation can be found here: https://docs.skriptlang.org/
[09:50:18] [Server thread/INFO]: [Skript] Skript's tutorials can be found here: https://docs.skriptlang.org/tutorials
[09:50:18] [Server thread/INFO]: [Skript] Server Version: git-Purpur-2014 (MC: 1.20.1)
[09:50:18] [Server thread/INFO]: [Skript] Skript Version: 2.7.0-beta3 (skriptlang-github)
[09:50:18] [Server thread/INFO]: [Skript] Installed Skript Addons: None
[09:50:18] [Server thread/INFO]: [Skript] Installed dependencies: 
[09:50:18] [Server thread/INFO]: [Skript]  - GriefPrevention v16.18.1
```

### Agreement

- [X] I have read the guidelines above and affirm I am following them with this report.

**Comment by Fusezion** [comment]
Should be duplicate of #5689

**Comment by AyhamAl-Ali** [comment]
Duplicate of #5689. Should be fixed by #5699
